### PR TITLE
Add ability to deep link into objects

### PR DESCRIPTION
**Pull request by digiguru**
I have a problem

In some Jira Instances I encounter they add custom fields to represent key data - such a Team Name.

The progress report allows us to group by team name.

However sometimes the setup for custom fields contains a nested objects  rather than plain values. Here's an example report....

My config holds the following
```    
Progress report epic team field: Team
Progress report epic deadline field: duedate
``` 

When running the API I can see the `Team` field is looking up a parameter called  `customfield_10800` - as it should - great

however on the report it has generated the following team names

image

Basically it doesn't know the name because it's a child of a property node in the JSON response.

### CODE REVIEW

File: jira_agile_metrics/querymanager.py [hunk 1]
```diff
@@ -8,6 +8,21 @@
 
 logger = logging.getLogger(__name__)
 
+def multi_getattr(obj, attr, **kw):
```
Comment (digiguru) [inline jira_agile_metrics/querymanager.py:11]: This function allows us to lookup an object using notation as follows 

Given the input
```
issue: {
    team: {
        name: "Mobile"
    }
}
```

we can reference the team name using ...
```
multi_getattr(issue, "team.name")
```

File: jira_agile_metrics/querymanager.py [hunk 2]
```diff
@@ -81,14 +96,17 @@ def __init__(self, jira, settings):
             self.fields_to_attributes[field_id] = name
 
     def field_name_to_id(self, name):
+        arr_name = name.split(".")
```
Comment (digiguru) [inline jira_agile_metrics/querymanager.py:99]: Because "team" actually gets translated to a customfield by the jira API. The old behaviour might change `team` to `customfield_10800`

This new code needs to translate `team.name` to `customfield_10800.name`

**Review by digiguru — COMMENTED** [review]
Walk through of the code

**Comment by digiguru** [comment]
As you can see this allows us to keep existing behaviour by default, but if you encounter a nested object in Jira you can simply edit the `yaml` file

```
Progress report epic team field: Team
Progress report epic deadline field: duedate
```
becomes 
```
Progress report epic team field: Team.name
Progress report epic deadline field: duedate
```

and you get the expected output

 
image

**Comment by digiguru** [comment]
Apologies for not providing a test class - I don't seem to be able to run the tests. Happy to help you through the review - please shout if I can improve it.
